pom.xml

```diff
@@ -51,6 +51,17 @@
             <artifactId>slf4j-api</artifactId>
             <version>1.7.36</version>
         </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-core</artifactId>
+            <version>5.3.22</version>
+        </dependency>
```

Comment:
we shouldn't pull in spring as a dependency here only because of cglib.
Instead, we should use bytebuddy or the java dynamic proxy.